contrib/endpoints/src/api_manager/path_matcher.h

```diff
@@ -129,6 +130,369 @@ class PathMatcherBuilder {
   friend class PathMatcher<Method>;
 };
 
+namespace {
+
+std::vector<std::string>& split(const std::string& s, char delim,
```

Comment:
These are in unnamed namespace and referred by those templated methods. If you have them in cc file the linker would complain. Having them outside of unnamed namespace would pollute the namespace.